californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -49,20 +49,36 @@
 public class InMemoryMessageExchangeStore implements MessageExchangeStore {
 
 	private static final Logger LOGGER = Logger.getLogger(InMemoryMessageExchangeStore.class.getName());
-	private static final int MAX_TOKEN_LENGTH = 8; // bytes
 	private final ConcurrentMap<KeyMID, Exchange> exchangesByMID = new ConcurrentHashMap<>(); // for all
 	private final ConcurrentMap<KeyToken, Exchange> exchangesByToken = new ConcurrentHashMap<>(); // for outgoing
 	private final ConcurrentMap<KeyUri, Exchange> ongoingExchanges = new ConcurrentHashMap<>();
 
 	private final NetworkConfig config;
-	private final int tokenLength;
 	private boolean running = false;
 	private Deduplicator deduplicator;
 	private ScheduledFuture<?> statusLogger;
 	private ScheduledExecutorService scheduler;
 	private MessageIdProvider messageIdProvider;
 	private TokenProvider tokenProvider;
 	private SecureRandom secureRandom;
+	
+	/**
+	 * Creates a new store for configuration values.
+	 * 
+	 * @param config the configuration to use.
+	 * 
+	 */
+	public InMemoryMessageExchangeStore(final NetworkConfig config) {
```

Comment:
simply do a `this(config, null)` and handle the `null` case appropriately in the other constructor?